src/kontrol/foundry.py

```diff
@@ -82,6 +94,75 @@
 _LOGGER: Final = logging.getLogger(__name__)
 
 
+class KontrolSemantics(KEVMSemantics):
+
+    def _check_forget_pattern(self, cterm: CTerm) -> bool:
+        """Given a CTerm, check if the rule 'FOUNDRY-ACCOUNTS.forget' is at the top of the K_CELL.
+
+        This method checks if the 'FOUNDRY-ACCOUNTS.forget' rule is at the top of the `K_CELL` in the given `cterm`.
+        If the rule matches, the resulting substitution is cached in `_cached_subst` for later use in `custom_step`
+        :param cterm: The CTerm representing the current state of the proof node.
+        :return: `True` if the pattern matches and a custom step can be made; `False` otherwise.
+        """
+        abstract_pattern = KSequence(
+            [
+                KApply('cheatcode_forget', [KVariable('###TERM1'), KVariable('###OPERATOR'), KVariable('###TERM2')]),
+                KVariable('###CONTINUATION'),
+            ]
+        )
+        self._cached_subst = abstract_pattern.match(cterm.cell('K_CELL'))
+        return self._cached_subst is not None
+
+    def _exec_forget_custom_step(self, cterm: CTerm, cterm_symbolic: CTermSymbolic) -> KCFGExtendResult | None:
+        """Given a CTerm, remove the constraint that's placed at the top of the K_CELL in the 'FOUNDRY-ACCOUNTS.forget' cut-rule.
+
+        :param cterm: CTerm of a proof node.
+        :param cterm_symbolic: CTermSymbolic instance
+        :return: If the K_CELL matches the forget_pattern and the constraint is successfully removed, returns a Step with depth 1 along with the new configuration, registering that the `cheatcode_forget` rule has been applied. If the constraint is not found, returns None.
```

Comment:
Obviously a nit, so feel free to ignore, but I think docstring lines are recommended to be <= 79 symbols.
```suggestion
        :return: If the K_CELL matches the forget_pattern and the constraint is 
            successfully removed, returns a Step with depth 1 along with the new 
            configuration, registering that the `cheatcode_forget` rule has been 
            applied. If the constraint is not found, returns None.
```